### ISSUE TITLE
[Distributed Index] Panic when restarting sync_gateway

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#1388]
Using Couchbase Sync Gateway/feature/distributed_index(807177a)

Panic - https://gist.github.com/sethrosetter/36410f970ab50f580643#file-sync_gateway_error-log-L1214. 

The scenario:
- 1 I was running a short gateload test against 2 sync_gateways (1 IW, 1 IR). 
- 2  As soon as gateload has completed, a shutdown is issued to the sync_gateways. (This is immediate so the writers may still be indexing) 
- 3 A couple minutes later, I started the writer and it crashed with the above panic.

**Comment by sethrosetter** [comment]
i discovered this when running gateload with the following command and config. 

```
python performance_tests/run_tests.py --use-gateload --gen-gateload-config --reset-sync-gw --number-pushers=500 --number-pullers=500
```

```
{
    "Hostname": "{{ sync_gateway_private_ip }}",
    "Port": 4984,
    "AdminPort": 4985,
    "Database": "db",
    "DocSize": 1024,
    "SendAttachment": false,
    "RampUpIntervalMs": 3000,
    "RunTimeMs": 30000,
    "SleepTimeMs": 10000,
    "NumPullers": {{ number_of_pullers }},
    "NumPushers": {{ number_of_pushers }},
    "ChannelActiveUsers": 40,
    "ChannelConcurrentUsers": 40,
    "MinUserOffTimeMs": 0,
    "MaxUserOffTimeMs": 0,
    "Verbose": false,
    "LogRequests": false,
    "UserOffset": {{ user_offset }},
    "AuthType": "basic",
    "Password": "password"
}
```

**Comment by sethrosetter** [comment]
Note: This is not reproducible every time. I ran the same scenario again and no crash. Perhaps increasing the load so indexing get backed up may help to reproduce this every time.

**Comment by adamcfraser** [comment]
My feeling is that this is related to #1371 - the initial shutdown is happening midway through a snapshot, and we're attempting to restart the DCP feed at an invalid point.  Should get retested/reviewed once #1371 is fixed.

**Comment by zgramana** [comment]
@sethrosetter please re-run on today's master.

**Comment by tleyden** [comment]
@sethrosetter see https://github.com/couchbase/sync_gateway/issues/1412#issuecomment-169482631 was build # 515

**Comment by sethrosetter** [comment]
@adamcfraser  @tleyden I was able to repo the panic targeting Couchbase Sync Gateway/1.2.0(35;a2a9c49). https://gist.github.com/sethrosetter/7d4604ae8f1b15fb9020#file-gistfile1-txt-L6

The steps to repro.
- Set up a cluster of 1 IW / 2 IR + 3 servers.
- Change gateload config to run for 10 min.
- Run with the below paramaters

```
python performance_tests/run_tests.py --use-gateload --gen-gateload-config --reset-sync-gw --number-pushers=2500 --number-pullers=2500 --test-id="5k-5k-2IR-1IW"
```
- Once the gateloads are done, they kill the sync_gateways as soon as they can
- ssh into the writer
- service sync_gateway start
- Crash

It looks like the tracking ticket for #1371 is targeted for 1.3. I'm bouncing this back to dev.

**Comment by adamcfraser** [comment]
Retargeting for 1.3, since this is addressed with #1371.

**Comment by adamcfraser** [comment]
Verify this when #1371 is complete.
